common/source/docs/common-network.rst

```diff
@@ -0,0 +1,118 @@
+
+In addition, LUA Scripting bindings allow the autopilot to run network server type applications via LUA scripts. A web server LUA example is available as an applet that allows faster log downloads, as well as vehicle/system status data to be presented.
+
+For peripherals with a network interface ArduPilot has AP_Periph firmware that allows the easy development of network peripheral devices using the peripheral's Serial or DroneCAN interfaces to the peripheral device and an easily set up means to pass those devices' data to the autopilot over the network.
+
+ArduPilot Network Framework Overview
+====================================
+
+ArduPilot provides a NET manager facility that provides the IP protocol engine, as well as UDP and TCP transport layer engines. Four configurable IP ports are available, and each one can be a UDP or TCP, Client or Server. Data flow to/from these ports can be used by the autopilot as if the data streams were those of a:
+
+- A serial device supported by ArduPilot (ie. Rangefinder, Gimbal, serial ESC, etc.)
+- A MAVLink connection to another vehicle or GCS
+
+In addition, LUA Scripting running on an autopilot can access these data ports.
+
+.. image:: ../../../images/Net_framework.jpg
+
+Configuration
+=============
+Global ArduPilot Configuration
+------------------------------
+
+- :ref:`NET_ENABLED<NET_ENABLED>` must be set to "1" and parameters refreshed in order to see the following:
+
+- :ref:`NET_MACADDR0<NET_MACADDR0>`
+- :ref:`NET_MACADDR1<NET_MACADDR1>`
+- :ref:`NET_MACADDR2<NET_MACADDR2>`
+- :ref:`NET_MACADDR3<NET_MACADDR3>`
+- :ref:`NET_MACADDR4<NET_MACADDR4>`
+- :ref:`NET_MACADDR5<NET_MACADDR5>`
+
+This is the global 6 byte MAC address. The default is automatically generated from the autopilot's CPU unique id, but can be changed if desired.
+
+- :ref:`NET_IPADDR0<NET_IPADDR0>`
+- :ref:`NET_IPADDR1<NET_IPADDR1>`
+- :ref:`NET_IPADDR2<NET_IPADDR2>`
+- :ref:`NET_IPADDR3<NET_IPADDR3>`
+
+This is the autopilot's 4 byte IP address. It defaults to ``192.168.13.14``. It can be dynamically assigned if the network has a DHCP server by enabling ArduPilot's DHCP client using the :ref:`NET_DHCP<NET_DHCP>` parameter, or manually set as desired.
+
+- :ref:`NET_NETMASK<NET_NETMASK>` is the number of leading bits set in the subnet mask. For example, if its set to 24, then the subnet mask is  255.255.255.0, which is the default.
+- :ref:`NET_GWADDR0<NET_GWADDR0>`
+- :ref:`NET_GWADDR1<NET_GWADDR1>`
+- :ref:`NET_GWADDR2<NET_GWADDR2>`
+- :ref:`NET_GWADDR3<NET_GWADDR3>`
+
+This is the gateway's IP address out of the local sub-net defined by the :ref:`NET_NETMASK<NET_NETMASK>` mask. Any destination IPs that are outside the masked subnet range will be sent to the gateway's MAC address with the final destinations IP address.
```

Comment:
Ditto the above comment about NET_ADDR.

Reply:
none of those appear on an autopilot that does not have an ENET MAC integrated.....

Reply:
Then perhaps that should be explained, as the first couple of sections _would_ apply even if there is no ethernet port - specifically when using ppp.

Reply:
Oh - I just noticed that statement doesn't seem to be true. all the NET_IPADDR and NET_GWADDR parameters do appear on my Durandal by just turning on NET_ENABLE. It has no integrated ethernet.

Reply:
hummm I just built Cube Orange on master with PPP and it has no NET_ENABLE param

Reply:
okay needs special config to build...now I see the params....also, it takes more than a reboot to enable some things, you have to power cycle

Reply:
Yes it's a bit confusing, some parameters changes 'work' with just a refresh, others require a reboot, I think I found there were a couple that won't even work on a soft reboot and require a hard/power off reboot.